lib/rubocop/cop/style/extra_spacing.rb

```diff
@@ -140,15 +140,17 @@ def unary_plus_non_offense?(range)
         end
 
         # Returns an array of ranges that should not be reported. It's the
-        # extra spaces between the keys and values in a hash, since those are
-        # handled by the Style/AlignHash cop.
+        # extra spaces between the keys and values in a multiline hash,
+        # since those are handled by the Style/AlignHash cop.
         def ignored_ranges(ast)
           return [] unless ast
 
           @ignored_ranges ||= on_node(:pair, ast).map do |pair|
+            next if pair.parent.single_line?
+
             key, value = *pair
             key.source_range.end_pos...value.source_range.begin_pos
-          end
+          end.compact
```

Comment:
I went with this `compact` approach rather than like do something like
```
on_node(:pair, ast).select { |pair| pair.parent.multiline? }
```

because I didn't want to turn the enumerator into an array.  Let me know if there is a cleaner way to do this! :)